widgets/secrets/src/UpdateModal.tsx

```diff
@@ -44,18 +43,25 @@ export default function UpdateModal({ open, secret, toolbox, onHide }: UpdateMod
         if (secret) {
             const isHidden = secret.is_hidden_value;
             if (hasSecretProvider) {
-                setUseSecretProvider(true);
-                setSecretProvider(secret.provider_name);
-                actions.doGet(secret.key).then(({ provider_options: providerOptions }) => {
-                    setSecretProviderOptions(providerOptions);
-                    setSecretProviderPath(providerOptions!.path);
-                });
+                setUseSecretProvider();
+                actions
+                    .doGetWithoutValue(secret.key)
+                    .then(({ provider_name: providerName, provider_options: providerOptions }) => {
+                        setSecretProvider(providerName);
+
+                        if (providerOptions) {
+                            const parsedSecretProvideOptions = JSON.parse(providerOptions);
+                            setSecretProviderPath(parsedSecretProvideOptions.path);
+                        }
```

Comment:
If `providerOptions` is a string which we have to parse, then probably safer would be to wrap `JSON.parse` with `try-catch` block.

```suggestion
                        if (providerOptions) {
                            try {
                                const parsedSecretProvideOptions = JSON.parse(providerOptions);
                                setSecretProviderPath(parsedSecretProvideOptions.path);
                            } catch (error) {
                                setErrors( . . . );
                            }
                        }
```

Ref.: https://stackoverflow.com/questions/4467044/proper-way-to-catch-exception-from-json-parse

Reply:
I even added try catch block earlier (https://github.com/cloudify-cosmo/cloudify-stage/pull/2427/commits/5163db5a5532e80dea7bbd55bb87da62aa37e813) :) but then modify it too many times so I lost it somewhere between commits - add it again: https://github.com/cloudify-cosmo/cloudify-stage/pull/2427/commits/e360c5395f2b5c874f7eec8c863a4d556c8cbbbc